### PR TITLE
Added support for Event Handling

### PR DESCRIPTION
**Pull request by raneja14**
TP can invoke addEvent method to create a custom
event based on business requirement. The arguments
to be passed should include 'event_type', 'attributes',
'event_data'

Signed-off-by: raghav.aneja <raghav.aneja@intel.com>

### CODE REVIEW

File: src/global_state.cpp [hunk 1]
```diff
@@ -129,5 +130,34 @@ void GlobalStateImpl::DeleteState(const std::vector<std::string>& addresses) con
...
+    request.set_context_id(this->context_id);
+    request.set_allocated_event(event);
+
+    FutureMessagePtr future = this->message_stream->SendMessage(
+        Message::TP_EVENT_ADD_REQUEST, request);
+    future->GetMessage(Message::TP_EVENT_ADD_RESPONSE, &response);
+
+    if(response.status() == TpEventAddResponse::ERROR){
+        std::stringstream error;
+        error << "Failed to add event";
```
Comment (agunde406) [inline src/global_state.cpp]: More information should be added to this error message, for example  event_type, kv_pairs,  and data.

Comment (raneja14) [reply src/global_state.cpp]: Have included event_type & event_data for the same

Comment (agunde406) [reply src/global_state.cpp]: The new commit is missing sign off and should just be squashed with the first commit instead.

Comment (raneja14) [reply src/global_state.cpp]: Done

**Pull request merged**